lib/iob/calculate.js

```diff
@@ -37,4 +58,76 @@ function iobCalc(treatment, time, dia) {
     return results;
 }
 
-exports = module.exports = iobCalc;
+
+function iobCalc(treatment, time, dia, profile) {
+
+	var curve = profile.curve;
+
+    if (curve === undefined || curve.toLowerCase() == 'bilinear') {
+    	return iobCalcBiLinear(treatment, time, dia);
+    };
+    
+    curve = curve.toLowerCase();
+    
+    if (curve != 'novorapid' && curve != 'fiasp') {
```

Comment:
I know I didn't include it in the code you started with, but it is probably worth including 'novolog' as a curve option  too.